lisa/tools/kdump.py

```diff
@@ -0,0 +1,206 @@
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Kexec, Makedumpfile]
+
+    @property
+    def command(self) -> str:
+        if isinstance(self.node.os, Redhat):
+            return "kdumpctl"
+        elif isinstance(self.node.os, Debian):
+            return "kdump-config"
+        elif isinstance(self.node.os, Suse):
+            return "kdumptool"
+        else:
+            posix_os: Posix = cast(Posix, self.node.os)
+            raise LisaException(
+                "Please check the accurate kdump tool command for distro "
+                f"{posix_os.name} {posix_os.information.version}."
+            )
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            self.node.os.install_packages("kexec-tools")
+        elif isinstance(self.node.os, Debian):
+            self.node.os.install_packages("linux-crashdump")
+        elif isinstance(self.node.os, Suse):
+            self.node.os.install_packages("kdump")
+
+        return self._check_exists()
+
+    def check_kernel_config(self, config_path: str) -> None:
+        for config in self.required_kernel_config:
+            result = self.node.execute(f"grep {config}=y {config_path}")
+            result.assert_exit_code(
+                message=f"The kernel config {config} is not set."
+                "Kdump is not supported."
+            )
+
+    def config_crashkernel_memory(
+        self,
+        crashkernel: str,
+    ) -> None:
+        # For Redhat and Suse, config crashkernel in /etc/default/grub
+        # For Ubuntu/Debian, the /etc/default/grub.d/kdump-tools.cfg will override the
+        # configuration in /etc/default/grub, so we config crashkernel in
+        # kdump-tools.cfg.
+        cfg_file = ""
+        if isinstance(self.node.os, Redhat):
+            if self.node.os.information.version < "8.0.0":
+                cfg_file = "/etc/default/grub"
+                if self.node.shell.exists(PurePosixPath("/sys/firmware/efi")):
+                    cmd = "grub2-mkconfig -o /boot/efi/EFI/redhat/grub.cfg"
+                else:
+                    cmd = "grub2-mkconfig -o /boot/grub2/grub.cfg"
+            else:
+                cmd = f'grubby --update-kernel=ALL --args="crashkernel={crashkernel}"'
+        elif isinstance(self.node.os, Suse):
+            cfg_file = "/etc/default/grub"
+            cmd = "grub2-mkconfig -o /boot/grub2/grub.cfg"
+        elif isinstance(self.node.os, Debian):
+            kdump_cfg = "/etc/default/grub.d/kdump-tools.cfg"
+            if self.node.shell.exists(PurePosixPath(kdump_cfg)):
+                cfg_file = kdump_cfg
+            else:
+                cfg_file = "/etc/default/grub"
+            cmd = "update-grub"
+
+        if cfg_file:
+            sed = self.node.tools[Sed]
+            result = self.node.execute(
+                f"grep crashkernel {cfg_file}", shell=True, sudo=True
```

Comment:
I think it doesn't need to use grep to check that. It can be done by a better regexp. If the line with `"GRUB_CMDLINE_LINUX_DEFAULT"`, it should works whether there is crashkernel or not.